test/ReactViews/Tools/ItemSearchTool/SearchResultsSpec.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { act, create, ReactTestRenderer } from "react-test-renderer";
+import BoundingSphere from "terriajs-cesium/Source/Core/BoundingSphere";
+import Cartesian3 from "terriajs-cesium/Source/Core/Cartesian3";
+import { ItemSearchResult } from "../../../../lib/Models/ItemSearchProvider";
+import SearchResults, {
+  ResultsCount,
+  SearchResultsProps
+} from "../../../../lib/ReactViews/Tools/ItemSearchTool/SearchResults";
+
+describe("SearchResults", function() {
+  it("shows the results count", async function() {
+    const { root } = await render({
+      results: randomResults(20),
+      onSelectResult: () => {}
+    });
+    const resultsCount = root.findByType(ResultsCount);
+    expect(resultsCount.props).toEqual(
+      jasmine.objectContaining({
+        count: 20
+      })
+    );
+  });
+});
+
+async function render(
+  props: Omit<SearchResultsProps, "i18n" | "t" | "tReady">
+): Promise<ReactTestRenderer> {
+  let rendered: ReactTestRenderer;
+  await act(() => {
+    rendered = create(<SearchResults {...props} />);
+  });
+  // @ts-ignore
+  return rendered;
+}
+
+function randomResults(count: number): ItemSearchResult[] {
```

Comment:
In my opinion, tests should be deterministic. Although using random test data every time covers more cases, it could mean that different test runs on the same code will return different results, which could be confusing. I would use this function to create a bunch of random data, save that to file, and then leave this function here but comment it out.

Reply:
Agreed about the predictability. I have changed `randomResults` to `sampleResults` after removing the randomness. Serializing an de-serializing from a file would be a bit too tedious as I have to manually re-construct the `BoundingSphere` instance.